#**Postwork Sesión 4: Análisis Exploratorio de nuestro Dataset**


###Equipo 23

* Francisco Ricardo Arredondo Almuina - frankk_arredondo@hotmail.com
* Tamara Ytanyu Pérez Vázquez - tamara.ytanyu@gmail.com
* Jesus Emanuel Serrano Molina - emanuel.serm@gmail.com
* Carlos Eduardo Vidal Villeda - charlie.lalo@hotmail.com


#Desarrollo
En este Postwork vamos a empezar a analizar nuestro conjunto de datos usando pandas. Sé que hasta ahora sólo hemos revisado cómo leer archivos tipo JSON usando pandas. El módulo está diseñado para aprender los diferentes tipos de formatos y explorar nuevas fuentes de datos en el momento en el que sean más apropiadas. En caso de que tu dataset tenga algún formato distinto a JSON, pídele ayuda a tu experto para que te guíe rápidamente en el proceso de lectura de tus datos. Tu experto puede proveerte código para que rápidamente puedas convertir tu conjunto de datos a un DataFrame. Algunas de las preguntas que estamos intentando responder en esta exploración son las siguientes:
Responde estas preguntas usando las técnicas que aprendiste en esta sesión y comparte tus hallazgos con tus compañeros y tu experto.

**¿El conjunto de datos que tengo realmente me sirve para responder algunas de las preguntas que me planteé?**

In [1]:
import pandas as pd
import numpy as np


In [3]:
#previamente se cargaron nuestros archivos csv a una cuenta de google drive
from google.colab import drive
drive.mount('/content/drive/')
!ls "/content/drive/MyDrive/Data_Proy"

Mounted at /content/drive/
Estatal-Delitos-2015-2021_ene2021.csv  Municipal-Delitos-2015-2021_ene2021.csv


In [4]:
df_mex_estados = pd.read_csv("/content/drive/MyDrive/Data_Proy/Estatal-Delitos-2015-2021_ene2021.csv", encoding="latin-1")
df_mex_municipios = pd.read_csv("/content/drive/MyDrive/Data_Proy/Municipal-Delitos-2015-2021_ene2021.csv", encoding="latin-1")

In [10]:
df_estados = pd.DataFrame(df_mex_estados)
df_estados.head(2)

Año  Clave_Ent         Entidad  ... Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...       2       2.0       1.0
1  2015          1  Aguascalientes  ...       0       0.0       1.0

[2 rows x 19 columns]

In [11]:
df_municipios = pd.DataFrame(df_mex_municipios)
df_municipios.head(2)

Año  Clave_Ent         Entidad  ...  Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...      1.0       0.0       1.0
1  2015          1  Aguascalientes  ...      0.0       0.0       0.0

[2 rows x 21 columns]

In [23]:
df_municipios.tail(2)

Año  Clave_Ent    Entidad  ...  Octubre Noviembre Diciembre
1588774  2021         32  Zacatecas  ...      NaN       NaN       NaN
1588775  2021         32  Zacatecas  ...      NaN       NaN       NaN

[2 rows x 21 columns]

_*Con un análisis rápido de nuestros data sets encotramos datos que incluyen información relevante a nuestra investigación, como lo son: tipos de delitos, estado de la República mexicana en donde ocurrieron y fecha en la que ocurrieron.
Por lo tanto podemos ser positivos de que nuestras preguntas serán respondidas*_


**¿Qué tamaño tiene mi conjunto de datos? ¿Serán datos suficientes?**

In [12]:
df_estados.shape

(21952, 19)

In [13]:
df_municipios.shape

(1588776, 21)

*Vemos que nuestro data frame **`df_estados`** tiene 19 columnas y 21952 filas,, mientras que el data frame **`df_municipios`** cuenta con 21 columnas y 1588776 filas, esta diferencia de filas es posible que se deba a que nuestro segundo data frame agrega la información en donde los actos de violencia fueron registrados (columna de municipios).
Debido a esto y a que ambos data sets fueron obtenidos del mismo lugar, en adelante solo se considerará **`df_municipios`** ya que la información está más completa agragando el campo de los municipios, mientras que **`df_estados`** es una recopilación de estos y estaríamos cometiendo un error de replica de datos*

**¿Qué columnas tengo y qué información tengo en cada una de esas columnas?**

In [14]:
df_municipios.columns

Index(['Año', 'Clave_Ent', 'Entidad', 'Cve. Municipio', 'Municipio',
       'Bien jurídico afectado', 'Tipo de delito', 'Subtipo de delito',
       'Modalidad', 'Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio',
       'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'],
      dtype='object')

- En la columna **`Año`** tenemos registros que empiezan en el 2015 y llegan hasta el 2021.
- En **`Entidad`** yacen los nombres de los estados donde ocurren los hechos mientras que **`Caleve_Ent`** es la calve federativa de este estado.
- Así en **`Municipio`** se agregan el municipio del estado correspondiente y **`Cve. Municipio`** es su clave única.
- **`Bien jurídico afectado`** , **`Tipo de delito`**, **`Subtipo de delito`** y **`Modalidad`** son columnas que describen los delitos registrados.
- Las columnas de **`Enero`**, **`Febrero`**, **`Marzo`**, **`Abril`**, **`Mayo`**, **`Junio`**, **`Julio`**, **`Agosto`**, **`Septiembre`**, **`Octubre`**, **`Noviembre`**, **`Diciembre`** son las sumas acumuladas de los registros por tipos de delito.


**Los nombres que tienen mis columnas, ¿son el nombre más apropiado?** 

- *Si, son descriptivas, simples y van de acuerdo a a la información que queremos obtener*

**¿Qué tipos de datos tengo en cada columna? ¿Parecen ser el tipo correcto de datos? ¿O es un tipo de datos "incorrecto"?** 

In [20]:
df_municipios.dtypes

Año                         int64
Clave_Ent                   int64
Entidad                    object
Cve. Municipio              int64
Municipio                  object
Bien jurídico afectado     object
Tipo de delito             object
Subtipo de delito          object
Modalidad                  object
Enero                       int64
Febrero                   float64
Marzo                     float64
Abril                     float64
Mayo                      float64
Junio                     float64
Julio                     float64
Agosto                    float64
Septiembre                float64
Octubre                   float64
Noviembre                 float64
Diciembre                 float64
dtype: object

- *Las columnas tienen el tipo de dato correcto, especialmente nos interesa que las columnas de fecha como `Año` y los casos acumulados sean datos de tipo númerico.*

**Si selecciono algunas filas al azar y las observo, ¿estoy obteniendo los datos que debería? ¿o hay datos que parecen estar "sucios" o "incorrectos"?** 

In [22]:
df_municipios.loc[[0,231,1112,45932,1588775], ['Año', 'Entidad', 'Municipio', 'Tipo de delito','Modalidad', 'Enero' ]]

Año  ... Enero
0        2015  ...     2
231      2015  ...     0
1112     2015  ...     4
45932    2015  ...     0
1588775  2021  ...     0

[5 rows x 6 columns]

- *Hasta el momento todo parece ser correcto, los datos no muestran estar sucios*